src/scripts/components/DateRangeDropdown.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Dropdown, DropdownMenuItem, DropdownMenuDivider, DropdownMenuItemWild, Date, Button, ResourceRight } from '.';
+
+class DateRangeDropdown extends React.Component {
+  state = {
+    isApplyEnabled: false,
+    startDate: moment(this.props.startDate),
+    endDate: moment(this.props.endDate),
+    minDate: moment(this.props.minDate),
+    maxDate: moment(this.props.maxDate),
+    dropdownMenuItems: this.props.dropdownMenuItems || [],
+    activeKey: this.props.activeKey,
+  };
+
+  componentDidMount() {
+    this.setState({ isApplyEnabled: false, startDate: moment(this.props.startDate), endDate: moment(this.props.endDate) });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.endDate !== this.state.endDate && this.props.isCustomDate) {
+      this.endDatePickerInput.datePickerInput.input.setRawValue(this.state.endDate.format('MM/DD/YYYY'));
```

Comment:
This conditional isn't too complicated, so it's not necessary here but consider moving the logic out of the if statement and into a descriptive constant declaration for easier readability

Reply:
@adlondon As per me we should use constant variables for condition check if we are adding same condition multiple times. If we will convert all the conditional logic into constant variable declaration then it will creates many variables. @adlondon @juanfabrega Your thoughts?